File: src/Fs.Units.CSharp/Simple/FuelEconomy.cs
// ReSharper disable once CheckNamespace
namespace Fs.Units.CSharp.FuelEconomy;

public static class MilePerGallonExtensions
{
    public static double MilePerGallonToKilometersPerLiter(this double x) => Units.FuelEconomy.MilePerGallon.toKilometersPerLiter(x);
}

public static class KilometerPerLiterExtensions
{
    public static double KilometerPerLiterToMilesPerGallon(this double x) => Units.FuelEconomy.KilometerPerLiter.toMilesPerGallon(x);
}

File: src/Fs.Units/Compound/FuelEconomy.fs
namespace Fs.Units

[<AutoOpen>]
module FuelEconomy =

  /// Kilometer per liter
  [<Measure>]
  type kml = km / L

  /// Mile per gallon
  [<Measure>]
  type mpg = mi / gal

  [<RequireQualifiedAccess>]
  module KilometerPerLiter =

    let create (x: float) = x * 1.0<kml>
    let toMilesPerGallon (x: float<kml>) = x * 2.35215<mpg> / 1.0<kml>

  [<RequireQualifiedAccess>]
  module MilePerGallon =

    let create (x: float) = x * 1.0<mpg>
    let toKilometersPerLiter (x: float<mpg>) = x / 2.35215<mpg> * 1.0<kml>
